libs/ui/src/screens/tokens/components/manage-tokens/manage-tokens.tsx

```diff
@@ -0,0 +1,89 @@
+import { isDefined, OnEventFn } from '@rnw-community/shared';
+import { isAddress } from 'ethers/lib/utils';
+import React, { FC, useEffect, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { Row } from '../../../../components/row/row';
+import { ScreenScrollView } from '../../../../components/screen-components/screen-scroll-view/screen-scroll-view';
+import { Switch } from '../../../../components/switch/switch';
+import { Token } from '../../../../components/token/token';
+import { Token as TokenInterface } from '../../../../interfaces/token.interface';
+import {
+  addNewTokenAction,
+  changeTokenVisibilityAction,
+  sortAccountTokensByVisibility
+} from '../../../../store/wallet/wallet.actions';
+import { useAccountTokensAndGasTokenSelector } from '../../../../store/wallet/wallet.selectors';
+import { checkIsGasToken } from '../../../../utils/check-is-gas-token.util';
+import { getTokenSlug } from '../../../../utils/token.utils';
+import { filterAccountTokensByValue } from '../../utils/filter-account-tokens-by-value.util';
+
+import { styles } from './manage-tokens.styles';
+
+interface Props {
+  searchValue: string;
+  newToken: TokenInterface | null;
+  setIsEmptyTokensList: OnEventFn<boolean>;
+}
+
+export const ManageTokens: FC<Props> = ({ searchValue, newToken, setIsEmptyTokensList }) => {
```

Comment:
I think this component have a lot in common with account tokens component. Maybe you can move up some similar things to general component?